src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -9,25 +9,28 @@
 
 namespace System.Text.Json.Serialization.Metadata
 {
+    /// <summary>
+    /// Provides JSON serialization-related metadata about a type.
+    /// </summary>
     [DebuggerDisplay("ClassType.{ClassType}, {Type.Name}")]
-    internal sealed partial class JsonTypeInfo
+    public partial class JsonTypeInfo
     {
-        public delegate object? ConstructorDelegate();
+        internal delegate object? ConstructorDelegate();
 
-        public delegate T ParameterizedConstructorDelegate<T>(object[] arguments);
+        internal delegate T ParameterizedConstructorDelegate<T>(object[] arguments);
 
-        public delegate T ParameterizedConstructorDelegate<T, TArg0, TArg1, TArg2, TArg3>(TArg0 arg0, TArg1 arg1, TArg2 arg2, TArg3 arg3);
+        internal delegate T ParameterizedConstructorDelegate<T, TArg0, TArg1, TArg2, TArg3>(TArg0 arg0, TArg1 arg1, TArg2 arg2, TArg3 arg3);
 
-        public ConstructorDelegate? CreateObject { get; private set; }
+        internal ConstructorDelegate? CreateObject { get; set; }
 
-        public object? CreateObjectWithArgs { get; set; }
+        internal object? CreateObjectWithArgs { get; set; }
 
         // Add method delegate for non-generic Stack and Queue; and types that derive from them.
-        public object? AddMethodDelegate { get; set; }
+        internal object? AddMethodDelegate { get; set; }
 
-        public ClassType ClassType { get; private set; }
+        internal ClassType ClassType { get; private set; }
```

Comment:
nit: also rename this? "ClassType" -> "ConverterStrategy"? Also perhaps this property can be removed since `classinfo.Converter.ClassType` (or `classinfo.Converter.ConverterStrategy` if renamed) could be used.

Reply:
Done. `ConverterStrategy` sounds good and describes exactly how the type is used. I removed `JsonTypeInfo.Converter` and `JsonTypeInfo.ClassType/ConverterStrategy`.  The way to get the strategy now is from `JsonTypeInfo.PropertyInfoForTypeInfo.ConverterStrategy`. I also removed `ConverterStrategy.NewValue` since it is not used.